dask_expr/_expr.py

```diff
@@ -417,7 +417,12 @@ def combine_similar(
 
             if changed_dependency:
                 expr = type(expr)(*new_operands)
-                changed = True
+                if isinstance(expr, Projection):
+                    # We might introduce stacked Projections (merge for example).
```

Comment:
Merge potentially introduces multiple Projections based on how the tree is constructed, we should squash those together again.

I am special casing for Projections for now, but I think there is no harm in opening this up to all Expressions as long as we don't call simplify_up. But I want more test coverage for this, so not doing it here

Reply:
It makes follow up steps more complicated if you have to start worrying about multiple projections stacked upon each other. This is more to reduce mental load than anything else. (It would have made parts of the combine_similar logic even more complicated).

Performance impact should ne negligible indeed.

Reply:
Okay, makes sense to me. Thanks for clarifying.